lib/src/update_version.dart

```diff
@@ -13,10 +13,45 @@ abstract class UpdateVersion extends Command {
 
   Future run() async {
     final dir = Directory(globalResults['pubspec-dir']);
-    final pubSpec = await PubSpec.load(dir);
-    final version = nextVersion(pubSpec.version);
-    await pubSpec.copy(version: version).save(dir);
-    console.log(version.toString());
+    final File pubSpec = File(p.join(dir.path, "pubspec.yaml"));
+
+    List<String> lines = pubSpec.readAsLinesSync();
+    String versionStringUnformatted =
+        lines.firstWhere((l) => l.startsWith("version: "), orElse: () => null);
+    if (versionStringUnformatted == null) {
+      throw new StateError("No version found!");
+    }
+
+    String versionString = _extractVersion(versionStringUnformatted);
+    Version oldVersion = Version.parse(versionString);
+    Version newVersion = nextVersion(oldVersion);
+    console.log(newVersion.toString());
+
+    _writeNewPubSpec(pubSpec, oldVersion, newVersion);
+  }
+
+  Future _writeNewPubSpec(
+      File pubSpec, Version oldVersion, Version newVersion) async {
+    String newPubSpec = pubSpec.readAsStringSync().replaceFirst(
+        'version: "${oldVersion}"', "version: " + '"${newVersion.toString()}"');
```

Comment:
What if a different quotation is used?
```yaml
version: '1.2.3+4'
```
or
```yaml
version: 1.2.3+4
```

Reply:
True, thanks. The ouput shouldn't matter but for the input I guess it's better to use a simple regex. I'll add it.